can/constructor-hydrate/constructor-hydrate.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * @module {connect.Behavior} can-connect/can/constructor-hydrate/constructor-hydrate
+ * @parent can-connect.behaviors
+ *
+ * Persists instances with the same id.
+ *
+ * @signature `constructorHydrate( baseConnection )`
+ *
+ * 	 Overrides [can-define/map/map]'s `setup` method and checks whether a newly created instance already
+ * 	 exists in [can-connect/constructor/store/store.instanceStore]. If it exists that instance will be
+ * 	 returned instead of a new object.
+ *
+ * 	 This behavior has to be used with [can-connect/constructor/store] and [can-connect/can/map/map] behaviors.
+ *
+ * @body
+ *
+ * ## Use
+ *
+ * This behavior could be useful if `Type` converters of [can-define/map/map] are used in multiple places of your app.
+ * In which case if a property is set with an id of an already created instance then the connection behaviour will
```

Comment:
the property isn't set to the `id` directly though right? It's not `{student: 5}`, it's `{student: {id: 5}}`.